gcf/custom/gcs.go

```diff
@@ -90,3 +91,30 @@ func FindFiles(ctx context.Context, c *storage.Client, bucket, pathPrefix string
 func BigStorePath(bucket, path string) string {
 	return filepath.Join("/bigstore", bucket, path)
 }
+
+// Find the root directory for data imports on gcs.
+// For data import directory structures, please refer to
+// https://docs.datacommons.org/custom_dc/upload_data.html
+//
+// Root directory in the above example is "gcs_folder"
+// Example:
+// findRootImportDirectory("/somepath/gcs_folder/internal/control/trigger.txt")
+// returns "/somepath/gcs_folder/"
+//
+// Note manifest generator requires the "/" at the end.
+func FindRootImportDirectory(triggerPath string) (string, error) {
```

Comment:
note: this has been moved to FindRootImportDirectory

Reply:
Done